kong/plugins/rate-limiting/handler.lua

```diff
@@ -104,6 +111,18 @@ function RateLimitingHandler:access(conf)
         ngx.header[RATELIMIT_LIMIT .. "-" .. k] = v.limit
         ngx.header[RATELIMIT_REMAINING .. "-" .. k] = math.max(0, (stop == nil or stop == k) and v.remaining - 1 or v.remaining) -- -increment_value for this current request
       end
+
+      -- Find the closer limit.
+      local closer_limit = 0
+      for k, v in pairs(usage) do
+        if (closer_limit == 0 or closer_limit > v.remaining) then
```

Comment:
This condition seems flawed to me.

1. It will find the "smallest limit", and not the "closest to quota" one.
2. If `v.remaining` is `0` in the first iteration (and any subsequent ones), then we enter the branch when in fact, the value is the same. While 1. is a bigger concern, this points just highlights inefficiency in the implementation of 1.

It is my belief that your intent is to show this header only for the closest limit anyway, so the condition to retrieve this limit needs to be reworked.

Reply:
- agree on 2. initializing  `closer_limit = -1` is better
- I want to just publish the lowest limit.

Reply:
done in ff678caf5. Feedback welcome ;)